internal/data_generator/internal/base_structs.go

```diff
@@ -78,6 +78,30 @@ func (es ${structName}) MoveTo(dest ${structName}) {
 	return
 }
 
+// CopyTo copies all elements from the current slice to the dest.
+func (es ${structName}) CopyTo(dest ${structName}) {
+	newLen := es.Len()
+	if newLen == 0 {
+		*dest.orig = []*${originName}(nil)
+		return
+	}
+	oldLen := dest.Len()
+	if newLen <= oldLen {
+		(*dest.orig) = (*dest.orig)[:newLen]
+		for i, el := range *es.orig {
+			new${elementName}(&el).CopyTo(new${elementName}(&(*dest.orig)[i]))
+		}
+		return
+	}
+	origs := make([]${originName}, newLen)
+	wrappers := make([]*${originName}, newLen)
+	for i, el := range *es.orig {
+		wrappers[i] = &origs[i]
+		new${elementName}(&el).CopyTo(new${elementName}(&wrappers[i]))
+	}
+    *dest.orig = wrappers
+}
+
 // Resize is an operation that resizes the slice:
 // 1. If newLen is 0 then the slice is replaced with a nil slice.
 // 2. If the newLen < len then equivalent with slice[0:newLen].
```

Comment:
Not totally sure of the reason for the change below as I assume that is a non-op if `newLen == oldLen`, but this comment should be updated to match

Reply:
It is a noop even after the change, the problem was that for "="  we went on the next path.